lib/zendesk_apps_support/manifest.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+module ZendeskAppsSupport
+  class Manifest
+    class << self
+      def private_attr_reader(*names)
+        private
+        attr_reader(*names)
+      end
+    end
+
+    LEGACY_URI_STUB = '_legacy'
+
+    RUBY_TO_JSON = {
+      name: 'name',
+      requirements_only: 'requirementsOnly',
+      version: 'version',
+      author: 'author',
+      framework_version: 'frameworkVersion',
+      single_install: 'singleInstall',
+      signed_urls: 'signedUrls',
+      no_template: 'noTemplate',
+      default_locale: 'defaultLocale',
+      original_locations: 'location',
+      private: 'private',
+      oauth: 'oauth',
+      original_parameters: 'parameters',
+      domain_whitelist: 'domainWhitelist',
+      remote_installation_url: 'remoteInstallationURL',
+      terms_conditions_url: 'termsConditionsURL',
+      google_analytics_code: 'gaID'
+    }
+
+    attr_reader(*RUBY_TO_JSON.keys)
+
+    alias_method :requirements_only?, :requirements_only
+    alias_method :signed_urls?, :signed_urls
+    alias_method :single_install?, :single_install
+    alias_method :private?, :private
+
+    def no_template?
+      if no_template.is_a?(Array)
+        false
+      else
+        no_template
+      end
+    end
+
+    def no_template_locations
+      no_template || []
+    end
+
+    def location?
+      locations != { 'zendesk' => {} }
+    end
+
+    def locations
+      @locations ||=
+        case original_locations
+        when Hash
+          original_locations
+        when Array
+          { 'zendesk' => NoOverrideHash[original_locations.map { |location| [ location, LEGACY_URI_STUB ] }] }
+        when String
+          { 'zendesk' => { original_locations => LEGACY_URI_STUB } }
+        # TODO: error out for numbers and Booleans
+        else # NilClass
```

Comment:
I'd prefer `when NilClass` (with an `else` which raises or explicitly ignores errors) rather than just having this be a comment.

Also is the above TODO supposed to be for now or later?